packages/ember-htmlbars/lib/keywords/get.js

```diff
@@ -0,0 +1,79 @@
+import Stream from "ember-metal/streams/stream";
+import isNone from 'ember-metal/is_none';
+import { labelFor } from "ember-metal/streams/utils";
+import { read, isStream } from "ember-metal/streams/utils";
+import create from "ember-metal/platform/create";
+import merge from "ember-metal/merge";
+import SafeString from "htmlbars-util/safe-string";
+
+export default function getKeyword(morph, env, scope, params, hash, template, inverse, visitor) {
+  var objParam = params[0];
+  var keyParam = params[1];
+
+  Ember.assert("The first argument to {{get}} must be a stream", isStream(objParam));
+  Ember.assert("{{get}} requires at least two arguments", params.length > 1);
+
+  var getStream = new GetStream(objParam, keyParam);
+
+  if (morph === null) {
+    return getStream;
+  } else {
+    env.hooks.inline(morph, env, scope, '-get', [getStream], params);
+  }
+
+  return true;
+}
+
+function workaroundValue(value) {
+  // this is used to compensate for the following line of code
+  //     if (result && result.value) {
+  // in the 'inline' hook. If the stream returns a falsy value after
+  // previously returning a truthy value, this line causes the value not
+  // to be updated
+
+  // so for falsey values we need to enforce something that evalutes to
+  // truthy but returns an empty string.
+
+  return isNone(value) ? new SafeString('') : value;
```

Comment:
The helper isn't invoked for subexpressions.. A basic example of where it fails is:

`context = { key: 'apple', colors: { apple: 'green', banana: 'yellow' } }`
`{{if trueValue (get colors key)}}` - outputs `green`
`context.set('key', null)`
`{{if trueValue (get colors key)}}` - outputs `green` still

Since `(get colors key)` once returned a true-like value but now returns a false-like value (`null`/`undefined`) the `inline` hook doesn't update the morph - but the desired behaviour is that it should - it should clear the value. This actually happens in the `inline`/`invokeHelper` for the `if` helper not the `get` keyword/helper.